src/uu/ls/src/ls.rs

```diff
@@ -332,9 +345,44 @@ struct PaddingCollection {
     major: usize,
     #[cfg(unix)]
     minor: usize,
+    block_size: usize,
 }
 
 impl Config {
+    /// Parse a SIZE string into a number of bytes.
+    /// A size string comprises an integer and an optional unit.
+    /// The unit may be k, K, m, M, g, G, t, T, P, E, Z, Y (powers of 1024), or b which is 1.
+    fn parse_byte_count(input: &str) -> Option<usize> {
```

Comment:
1. Yes, it would be invalid input.  When I tested this awhile ago on what I assume was an earlier version of coreutils, I believe it would silently fail.  I presumed a better method was to work around the bad input and parse what we could.  Now, gls 9.0 seems to note an error.  It may be that I'm remembering wrong, but yes, I think failing with an error is better behavior.
2. 9 kB is allowed here.  It's just parsed as 9k which is fine for parse_size to deal with.
3. I'll try to make it better match GNU behavior.

Reply:
I've removed this fn and we just use the parse_size fn now.  Thanks.